sound/pci/hda/patch_hdmi.c

```diff
@@ -120,6 +120,12 @@ struct hdmi_pcm {
 	struct snd_kcontrol *eld_ctl;
 };
 
+enum {
+	SILENT_STREAM_OFF = 0,
+	SILENT_STREAM_KAE,
+	SILENT_STREAM_I915
```

Comment:
a comment on why you have three definitions would help, I wasn't able to follow what that means later in the code. Specifically it's not clear what the _I915 variant might be.

Reply:
or is this legacy with an HDaudio transfer of zeroed data to the iDisp codec?

Reply:
Thanks @plbossart , exactly the kind of feedback I'm looking for. I'll add comments. Pre-ADL hw has its own keep-alive method that is not standard. We don't actually send zeroed data, but with a special hw sequence we can trigger the i915 to send data to the display link even when audio DMA is not sending anything. This is SILENT_STREAM_I915 and what was added to the codec driver last year.